keanu-project/src/main/java/io/improbable/keanu/algorithms/sampling/ForwardSampler.java

```diff
@@ -0,0 +1,52 @@
+package io.improbable.keanu.algorithms.sampling;
+
+import io.improbable.keanu.KeanuRandom;
+import io.improbable.keanu.algorithms.NetworkSample;
+import io.improbable.keanu.algorithms.Variable;
+import io.improbable.keanu.algorithms.VariableReference;
+import io.improbable.keanu.algorithms.mcmc.SamplingAlgorithm;
+import io.improbable.keanu.vertices.NonProbabilistic;
+import io.improbable.keanu.vertices.Probabilistic;
+import io.improbable.keanu.vertices.Vertex;
+
+import java.util.List;
+import java.util.Map;
+
+public class ForwardSampler implements SamplingAlgorithm {
+
+    private static final double LOG_PROB_OF_PRIOR = 0.;
```

Comment:
Why is this 0?

Reply:
See conversation between Chris and Caleb above

Reply:
As discussed - would be good to raise a ticket to make tracking of sample log probs optional across the board and implement the true log prob calculation here if someone turns it on.

Reply:
it needs some explanation in the code, especially since below you do `logOfMasterPForEachSample.add(LOG_PROB_OF_PRIOR)`